app/main.py

```diff
@@ -1,3 +1,81 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: object) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
+
+    def __iadd__(self, other: object) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
+        return self
+
+    def __mul__(self, other: object) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
+
+    def __truediv__(self, other: object) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        elif isinstance(other, (int, float)):
+            return self.km == other
+        return NotImplemented
+
+    def __lt__(self, other: object) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        elif isinstance(other, (int, float)):
+            return self.km < other
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
+
+    def __le__(self, other: object) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        elif isinstance(other, (int, float)):
+            return self.km <= other
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
+
+    def __gt__(self, other: object) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        elif isinstance(other, (int, float)):
+            return self.km > other
+        return NotImplemented
+
+    def __ge__(self, other: object) -> bool:
+        if isinstance(other, Distance):
+            return self.km >= other.km
+        elif isinstance(other, (int, float)):
+            return self.km >= other
+        return NotImplemented
+
+    def __ne__(self, other: object) -> bool:
+        if isinstance(other, Distance):
+            return self.km != other.km
+        else:
+            raise TypeError(f"Unsupported type: {type(other)}")
```

Comment:
The `__ne__` method only supports comparison with `Distance` objects and raises a `TypeError` for other types. This is inconsistent with other comparison methods like `__eq__`, which also handle `int` and `float` types. Consider updating `__ne__` to handle these types similarly.